pandas/indexes/category.py

```diff
@@ -574,6 +574,32 @@ def append(self, other):
         codes = np.concatenate([c.codes for c in to_concat])
         return self._create_from_codes(codes, name=name)
 
+    def _array_putmask(self, a, mask, values):
+        """
+        Equivalent to np.putmask(a, mask, values).
+
+        "a.categories" must already contain all categories that "values" uses.
+        """
+
+        rep_vals = a.categories.get_indexer(values.categories
+                                            ).view(dtype=a.codes.dtype)
+
+        idces = np.digitize(values.codes, np.arange(len(values.categories)),
```

Comment:
To my understanding (which is probably very limited), never in the entire code base do we do anything combining ``Categorical``s with different categories. Am I wrong? Or are you just suggesting I should move this in ``Categorical`` (then yes, I agree with you... but I would still use ``np.digitize``).